src/tests/test_pregnancy.py

```diff
@@ -0,0 +1,385 @@
+            assert mean - 3 * stdev <= no_pregnant <= mean + 3 * stdev
+            # low fertility individuals don't become pregnant
+            assert sum(pop.data[col.LOW_FERTILITY] & pop.data[col.PREGNANT]) == 0
+
+
+def test_stp_preg():
+
+    test_ages = [20, 30, 40, 50]
+    test_stp_partners = [1, 3, 5]
+    for age in test_ages:
+        for stp in test_stp_partners:
+
+            # build artificial population
+            N = 100000
+            pop = Population(size=N, start_date=date(1990, 1, 1))
+            pop.data[col.SEX] = SexType.Female
+            pop.data[col.AGE] = age
+            pop.data[col.NUM_PARTNERS] = stp
+            pop.data[col.LONG_TERM_PARTNER] = False
+            pop.data[col.LAST_PREGNANCY_DATE] = None
+            pop.data[col.NUM_CHILDREN] = 0
+            pop.data[col.WANT_NO_CHILDREN] = False
+            pop.pregnancy.prob_pregnancy_base = 0.1
+            pop.pregnancy.init_fertility(pop)
+            pop.pregnancy.update_pregnancy(pop)
+
+            # get stats
+            no_active_female = sum((~pop.data[col.LOW_FERTILITY])
+                                   & ((pop.data[col.NUM_PARTNERS] > 0)
+                                      | pop.data[col.LONG_TERM_PARTNER]))
+            no_pregnant = sum(pop.data[col.PREGNANT])
+            prob_preg = pop.pregnancy.calc_prob_preg(test_ages.index(age)+1, False, stp, False)
+            mean = no_active_female * prob_preg
+            stdev = sqrt(mean * (1 - prob_preg))
+            # check pregnancy value is within 3 standard deviations
+            assert mean - 3 * stdev <= no_pregnant <= mean + 3 * stdev
+
+
+def test_childbirth():
+
+    N = 100
+    time_step = timedelta(days=90)
+
+    # build artificial population
+    pop = Population(size=N, start_date=date(1990, 1, 1))
+    pop.data[col.SEX] = SexType.Female
+    pop.data[col.AGE] = 18
+    pop.data[col.LOW_FERTILITY] = False
+    pop.data[col.NUM_PARTNERS] = 0
+    pop.data[col.LONG_TERM_PARTNER] = True
+    pop.data[col.LAST_PREGNANCY_DATE] = None
+    pop.data[col.NUM_CHILDREN] = 0
+    # guaranteed pregnancy
+    pop.pregnancy.prob_pregnancy_base = 1
+
+    # evolve population
+    for i in range(0, ceil(timedelta(days=270)/time_step)):
+        # advance pregnancy
+        pop.pregnancy.update_pregnancy(pop)
+        pop.date += time_step
+
+    # final advancement into childbirth
+    pop.pregnancy.update_pregnancy(pop)
+    pop.date += time_step
+    # check that nobody is pregnant anymore
+    assert ~pop.data[col.PREGNANT].all()
+    # check that everyone now has one child
+    assert (pop.data[col.NUM_CHILDREN] == 1).all()
+
+    # test the pregnancy pause period
+    pop.pregnancy.update_pregnancy(pop)
+    pop.date += time_step
+    # check that there are still no pregnancies
+    assert ~pop.data[col.PREGNANT].all()
+
+    pop.pregnancy.update_pregnancy(pop)
+    pop.date += time_step
+    # check that everyone is pregnant again
+    assert pop.data[col.PREGNANT].all()
```

Comment:
I am a bit confused about this, has enough time passed for people to become pregnant again?